client/src/main/java/com/battleships/gamewindow/board/PlayerBoard.java

```diff
@@ -1,45 +1,52 @@
 package com.battleships.gamewindow.board;
 
 import com.battleships.Coordinate;
-import com.battleships.gamewindow.board.fieldStates.*;
+import com.battleships.gamewindow.board.fieldStates.BoardField;
+import com.battleships.gamewindow.board.fieldStates.FieldState;
 import com.battleships.gamewindow.services.RandomFleetPlacement;
-import javafx.scene.layout.GridPane;
 
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class PlayerBoard extends Board {
+    private final Fleet fleet;
     private final RandomFleetPlacement randomFleetPlacement;
 
     public PlayerBoard() {
         this.randomFleetPlacement = new RandomFleetPlacement();
+        fleet = new Fleet();
     }
 
-    public void changeAllFieldsToSea(BoardSize boardSize, GridPane gridPaneForBoard) {
-        for (int row = 1; row <= boardSize.rowsAmount(); row++) {
-            for (int col = 1; col <= boardSize.colAmount(); col++) {
-                Coordinate cord = Coordinate.fromIntCoords(row, col);
-                BoardField seaField = new BoardField(cord, FieldState.SEA);
-                gridPaneForBoard.add(seaField, col, row);
-                board.put(cord, seaField);
-            }
-        }
+    public void changeAllFieldsToSea() {
+        board.keySet().forEach(coordinate -> board.get(coordinate).setFieldState(FieldState.SEA));
     }
 
-    public void placeFleetRandomly (GridPane gridPaneBoard) {
-        List<Coordinate> shipsFleetCoords = randomFleetPlacement.getRandomCoords();
-        for (Coordinate mastCoord : shipsFleetCoords) {
-            BoardField boardField = new BoardField(mastCoord, FieldState.MAST);
-
-            int fieldColumn = mastCoord.getColumn();
-            int fieldRow = mastCoord.getRow();
-
-            this.board.put(mastCoord, boardField);
-            gridPaneBoard.add(boardField, fieldColumn, fieldRow);
+    public void placeFleetRandomly() {
+        List<Coordinate[]> fleetCoords = randomFleetPlacement.getRandomCoords();
+        fleet.clear();
+        for (Coordinate[] shipCoordinates : fleetCoords) {
```

Comment:
I would extract private methods from that two for loops. It would look clearer and intention would be much simpler to get to know.

Reply:
yeap, avoid create "for in for". You have better extract for to separate method